README.md

```diff
@@ -10,18 +10,47 @@ Usage
 ---------------------
 
 ```
-  xortool [-h|--help] [OPTIONS] [<filename>]
+xortool
+  A tool to do some xor analysis:
+  - guess the key length (based on count of equal chars)
+  - guess the key (base on knowledge of most frequent char)
+
+Usage:
+  xortool [-x] [-m MAX-LEN] [-f] [-t CHARSET] [FILE]
+  xortool [-x] [-l LEN] [-c CHAR | -b | -o] [-f] [-t CHARSET] [FILE]
+  xortool [-x] [-m MAX-LEN| -l LEN] [-c CHAR | -b | -o] [-f] [-t CHARSET] [FILE]
+  xortool [-h | --help]
+  xortool --version
+
 Options:
-  -l,--key-length       length of the key (integer)
-  -c,--char             most possible char (one char or hex code)
-  -m,--max-keylen=32    maximum key length to probe (integer)
-  -x,--hex              input is hex-encoded str
-  -b,--brute-chars      brute-force all possible characters
-  -o,--brute-printable  same as -b but will only use printable
-                        characters for keys
+  -x --hex                          input is hex-encoded str
+  -l LEN, --key-length=LEN          length of the key
+  -m MAX-LEN, --max-keylen=MAX-LEN  maximum key length to probe [default: 65]
+  -c CHAR, --char=CHAR              most frequent char (one char or hex code)
+  -b --brute-chars                  brute force all possible most frequent chars
+  -o --brute-printable              same as -b but will only check printable chars
+  -f --filter-output                filter outputs based on the charset
+  -t CHARSET --text-charset=CHARSET target text character set [default: printable]
+  -h --help                         show this help
+
+Notes:
+  Text character set:
+    * Pre-defined sets: printable, base32, base64
```

Comment:
I think there should be "all" charset defined (even it it is equivalent to no filtering).